zanata-war/src/main/java/org/zanata/service/impl/TranslationFileServiceImpl.java

```diff
@@ -282,24 +287,35 @@ private boolean hasAdapterFor(String fileNameOrExtension) {
     }
 
     private FileFormatAdapter getAdapterFor(String fileNameOrExtension) {
-        String extension = extractExtension(fileNameOrExtension);
+        String extension = FileUtil.extractExtension(fileNameOrExtension);
         if (extension == null) {
             throw new RuntimeException(
                     "Cannot find adapter for null filename or extension.");
         }
-        DocumentType documentType = DocumentType.typeFor(extension);
+        DocumentType documentType = DocumentType.getByName(extension);
         if (documentType == null) {
             throw new RuntimeException(
-                    "Cannot choose an adapter because the provided string '"
-                            + fileNameOrExtension
-                            + "' does not match any known document type.");
+                "Cannot choose an adapter because the provided string '"
+                    + fileNameOrExtension
+                    + "' does not match any known document type.");
+        }
+        FileFormatAdapter adapter = getAdapterFor(documentType);
+        if(hasMultipleAdapter(fileNameOrExtension)) {
+            /**
+             * TODO: throw runtime error. Need to wait for all upload file
+             * dialog implement multiple adapter check for file extension.
+             */
```

Comment:
- [ ] Why does it have to wait? Is it because it would be impossible for properties files to be uploaded at the moment? I would expect the requirement is that the dialogs allow the user to choose an adapter.